From 985d1f500b1214c7155c734d7e8df08dc2d0454b Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 May 2026 13:30:05 +0800
Subject: [PATCH] arm: arch-aspeed: add AST2705 platform definitions

Extend the platform header to cover CONFIG_ASPEED_AST2705 alongside
AST2700. Add ASPEED_IO_XGMAC_BASE for the 10G MAC at 0x140a0000.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I1771eacfd95bce026eeae2a5ce8cc6ae655a4421
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/include/asm/arch-aspeed/platform.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/include/asm/arch-aspeed/platform.h b/arch/arm/include/asm/arch-aspeed/platform.h
index 2a7c3e3293..6d65f6f0c5 100644
--- a/arch/arm/include/asm/arch-aspeed/platform.h
+++ b/arch/arm/include/asm/arch-aspeed/platform.h
@@ -55,7 +55,7 @@
 #define ASPEED_SRAM_BASE	0x10000000
 #define ASPEED_SRAM_SIZE	0x16000
 #define ASPEED_FMC_CS0_BASE	0x20000000
-#elif defined(CONFIG_ASPEED_AST2700)
+#elif defined(CONFIG_ASPEED_AST2700) || defined(CONFIG_ASPEED_AST2705)
 #define ASPEED_CPU_AHBC_BASE	0x12000000
 #define ASPEED_CPU_REVISION_ID	0x12C02000
 #define ASPEED_CPU_SCU_BASE	0x12C02000
@@ -72,6 +72,7 @@
 #define ASPEED_FMC_CS0_SIZE	0x80000000
 #define ASPEED_IO_MAC0_BASE	0x14050000
 #define ASPEED_IO_MAC1_BASE	0x14060000
+#define ASPEED_IO_XGMAC_BASE	0x140a0000
 #define ASPEED_IO_AHBC_BASE	0x140b0000
 #define ASPEED_IO_REVISION_ID	0x14C02000
 #define   CHIP_AST2700A1_ID_MASK      BIT(16)
-- 
2.43.0

